src/base/utils/sys.cpp

```diff
@@ -0,0 +1,218 @@
+                setTorrentFileAssoc();
+                setMagnetLinkAssoc();
+            }
+            else {
+                Preferences::instance()->setNeverCheckFileAssoc();
+            }
+        }
+    }
+
+#ifdef APP_ASSOC_REG
+    void showInitialAssocUIImpl2()
+    {
+        const QString dlgTitle {tr("Torrent file association")};
+        const QString dlgText {
+            tr("qBittorrent is not the default application to open torrent files or Magnet links.\n"
+               "Do you want to associate qBittorrent to torrent files and Magnet links?")
+        };
+
+        HRESULT hr = ::CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+        if (FAILED(hr)) return;
+
+        IApplicationAssociationRegistration *pAssoc;
+
+        hr = ::CoCreateInstance(
+                 CLSID_ApplicationAssociationRegistration, 0, CLSCTX_INPROC
+                 ,__uuidof(IApplicationAssociationRegistration), reinterpret_cast<void**>(&pAssoc));
+        if (SUCCEEDED(hr)) {
+            BOOL isDefault = TRUE;
+
+            hr = pAssoc->QueryAppIsDefaultAll(AL_EFFECTIVE, APP_REGISTRY_NAME, &isDefault);
+            if ((SUCCEEDED(hr) && (isDefault == FALSE)) || (hr == HRESULT_FROM_WIN32(ERROR_NO_ASSOCIATION))) {
+                if (QMessageBox::question(nullptr, dlgTitle, dlgText, QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes) == QMessageBox::Yes)
+                    pAssoc->SetAppAsDefaultAll(APP_REGISTRY_NAME);
+                else
+                    Preferences::instance()->setNeverCheckFileAssoc();
+            }
+
+            pAssoc->Release();
+        }
+
+        ::CoUninitialize();
+    }
+
+    void showInitialAssocUIImpl3()
+    {
+        const QString dlgTitle {tr("Torrent file association")};
+        const QString dlgText {
+            tr("Do you want to set qBittorrent as default program for .torrent files\n"
+               "and Magnet links?\n"
+               "You can do it later using \"Default Programs\" dialog from Control Panel.")
+        };
+
+        if (QMessageBox::question(nullptr, dlgTitle, dlgText, QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes) == QMessageBox::Yes)
+        {
+            Utils::Sys::showAppAssocRegUI();
+        }
+
+        Preferences::instance()->setNeverCheckFileAssoc();
+    }
+#else
+    void showInitialAssocUIImplFallback()
+    {
+        const QString dlgTitle {tr("Torrent file association")};
+        const QString dlgText {
+            tr("To set qBittorrent as default program for .torrent files and/or Magnet links\n"
+               "you can use \"Default Programs\" dialog from Control Panel.")
+        };
+
+        QMessageBox::information(nullptr, dlgTitle, dlgText, QMessageBox::Ok);
+        Preferences::instance()->setNeverCheckFileAssoc();
+    }
+#endif
+}
+
+bool Utils::Sys::isTorrentFileAssocSet()
```

Comment:
`isTorrentFileAssocSet()` and `isMagnetLinkAssocSet()` are logically identical and have to be replaced with a single function with parameter(s). Or simply refactor the common code out into a _imp function.